### PR TITLE
refactor(demo): update navigation and demo

### PR DESCRIPTION
**Pull request by CITguy**
(a.k.a. Demo Reorganization - part 2)

* Demo navigation now follows layout expected from Design
* lots of reorganization of demo code
  * many examples added/moved
  * many templates added/moved
  * `demoApp` now only includes `.config()` and `.run()`
    * additional logic moved to appropriate locations
* component:progressbar moved to Molecules
* component:grids docs removed and content merged with layout:grids
* `encore.ui.molecules` depends on `encore.ui.quarks` and `encore.ui.atoms`
* `encore.ui.atoms` depends on `encore.ui.quarks`
* modules in search and nav now sorted by their display name
  * allows easier organization later on
* update TravisCI to use node 4.2.3 LTS
* support `.node-version` locally
* removed unnecessary documentation
  * Overview > EncoreUI Design section
  * Styleguide > Basics page

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM

Screenshots to follow.

### CODE REVIEW

**Comment by CITguy** [comment]
## Overview Page
overview-after
* "EncoreUI Design" section removed
* Updated navigation menu

**Comment by CITguy** [comment]
## Layout > Grids
layout-grids-after
* Big red warning of deprecation
* `grid` component documentation merged
* `grid` component removed

**Comment by CITguy** [comment]
## Styleguide > Formatting
styleguide-formatting-after
* *New Page!*

**Comment by CITguy** [comment]
## Styleguide > Helper Classes
styleguide-helperclasses-after
* *New Page!*

**Comment by CITguy** [comment]
### Molecules > Progress Bars
molecules-progressbars-after
* Moved from Components
* No API link (no internal logic to document)

**Comment by CITguy** [comment]
### Atoms > Buttons
atoms-buttons-after

### Atoms > Links
atoms-links-after
* Separated from old Buttons documentation

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 sudo: false
 language: node_js
 node_js:
-- '0.12'
+  - "4.2.3" # same as .node-version
```
Comment (CITguy) [inline .travis.yml]: Updated to latest NodeJS LTS release

File: demo/less/demoApp.less [hunk 2]
```diff
@@ -66,18 +66,7 @@
   .readme-title {
     margin-top: 0;
   }
-}//.component-description
-
-.styleguide-character-count {
```
Comment (CITguy) [inline demo/less/demoApp.less]: unused class

File: demo/less/demoApp.less [hunk 3]
```diff
@@ -149,30 +138,6 @@ p {
...
-  h4,
-  h5,
-  h6 {
-    &:first-child {
-      margin-top: 0;
-    }
-  }
-}
-
-rx-page.styleguide {
```
Comment (CITguy) [inline demo/less/demoApp.less]: unused class, no longer needed

File: demo/scripts/demoApp.js [hunk 4]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
             $window.scrollTo(0,0);
         }
     });
-})
-
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: Controllers below were moved to their respective examples.

File: demo/scripts/demoApp.js [hunk 5]
```diff
@@ -463,133 +384,4 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
...
-    };
-
-    $scope.clickMe = function () {
-        $scope.status.loading = true;
-        $timeout(function () {
-            $scope.status.loading = false;
-        }, 4000);
-    };
-})
-.controller('commentBoxStyleguideCtrl', function ($scope) {
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: unused, removed

File: demo/templates/styleguide/basics.html [hunk 6]
```diff
@@ -1,161 +0,0 @@
...
-                      <code>.clear-left</code> and <code>.clear-right</code>
-                      classes.
-                    </li>
-                </ul>
-            </li>
-        </ul>
-    </div>
-    <rx-styleguide code-url="templates/styleguide/basics/helpers.html"></rx-styleguide>
-
-    <h3 id="progress-bars">Progress Bars</h3>
```
Comment (CITguy) [inline demo/templates/styleguide/basics.html]: moved to `Molecules > Progress Bars`

File: demo/templates/styleguide/basics.html [hunk 7]
```diff
@@ -1,161 +0,0 @@
...
-          extend too far into the past. For instance, "1 week ago" may be
-          helpful to the user, but "8 months ago" or "3 years ago" is better
-          represented by the non-relative format detailed above, as it reveals
-          more specific information about dates that provide better context and
-          information to the user when comparing dates and times.
-        </li>
-    </ul>
-
-    <h2 id="helper-classes">Helper Classes</h2>
-    <div class="module-description clear">
```
Comment (CITguy) [inline demo/templates/styleguide/basics.html]: moved to `Styleguide > Helper classes`

File: demo/templates/styleguide/basics.html [hunk 8]
```diff
@@ -1,161 +0,0 @@
...
-          </a>. With minor usability enhancements, EncoreUI provides an
-          assortment of attribute-based layout options, based on the flexbox
-          layout model. Included are intuitive attribute based styles for
-          responsive row and/or column based page layouts. Visit the
-          <a href="#/components/layout">layout</a> section for info on how to
-          use these attributes.
-        </li>
-    </ul>
-
-    <h3 id="date-formatting">Date formatting</h3>
```
Comment (CITguy) [inline demo/templates/styleguide/basics.html]: moved to `Styleguide > Formatting`

File: demo/templates/styleguide/basics.html [hunk 9]
```diff
@@ -1,161 +0,0 @@
...
-        <li>
-          Kevin Lamping, a developer on the EncoreUI team, wrote this document
-          to help you learn more about
-          <a href="https://github.com/rackerlabs/encore-ui/blob/master/guides/css-styleguide.md">
-            CSS coding standards
-          </a>.
-        </li>
-    </ul>
-
-    <h3 id="flexbox-grid">Flexbox Grid</h3>
```
Comment (CITguy) [inline demo/templates/styleguide/basics.html]: removed, already discussed in `Layout > Grid`

File: demo/templates/styleguide/basics.html [hunk 10]
```diff
@@ -1,161 +0,0 @@
-<rx-page title="'Introduction'">
-    <h2 class="clear">Basics</h2>
-    <ul class="list">
```
Comment (CITguy) [inline demo/templates/styleguide/basics.html]: remainder removed, majority of content already linked, mentioned, or moved

**Comment by glynnis** [comment]
This LGTM. Looks like we might need to fix some spacing between headers (h1 vs. h2 looks like), but I think that can be addressed at a later time as more of the pages and documentation get shifted around.

**Comment by 100stacks** [comment]
@CITguy `Atoms` and `Molecules` have different breadcrumb nomenclature  in https://github.com/rackerlabs/encore-ui/pull/1452#issuecomment-164556468 -- 

* `Overview > Buttons`
screen shot 2015-12-15 at 3 43 23 pm


* `Overview > Links`
screen shot 2015-12-15 at 3 43 38 pm


* `Overview > Modules > Molecules > Progress Bars`.  
screen shot 2015-12-15 at 3 35 00 pm


Also, is `Molecules` a child of `Modules` to indicate `Molecules` are/can be combined of subcomponents?

**Comment by 100stacks** [comment]
@CITguy @glynnis based on feedback from the sprint retro the breadcrumbs are something we'll sort out after navigation and demo migration.

LGTM :+1:

File: demo/examples/table.filtering.html [hunk 11]
```diff
@@ -0,0 +1,22 @@
+<table class="table-striped" ng-controller="tableFilteringExampleCtrl">
+  <thead>
+    <tr><!-- Filter Row -->
+      <td colspan="2">
+        <rx-search-box
+          rx-placeholder="'Filter by any...'"
+          class="filter-box"
+          ng-model="searchText"></rx-search-box>
```
Comment (parlarjb) [inline demo/examples/table.filtering.html]: Can we put `</rx-search-box>` on a new line?

File: demo/templates/atoms/buttons.html [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+  <h3>Button Sizing</h3>
+  <p>
+    Buttons are sized using xl, lg, sm and xs classes. Not adding the additional
+    class defaults the size to medium.
+  </p>
+  <rx-example name="button.sizing"></rx-example>
+
+
+  <h3>Button Types & Using Color</h3>
```
Comment (parlarjb) [inline demo/templates/atoms/buttons.html]: Use `&amp;` instead?

File: demo/templates/atoms/links.html [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+      When part of a list, they appear last.
+    </li>
+  </ul>
+  <rx-example name="link.colors"></rx-example>
+
+  <h2>Using Spinners with Action Links</h2>
+  <p>
+    You can use a spinner style for action links. Different spinner styles are
+    available when a user takes action by clicking a button
+    (see <a href="#">Buttons & Links demo</a>).
```
Comment (parlarjb) [inline demo/templates/atoms/links.html]: `&amp;`

**Comment by 100stacks** [comment]
Created tech debt story to capture breadcrumbs and navigation differences.  https://jira.rax.io/browse/FRMW-557

**Comment by CITguy** [comment]
@parlarjb, feedback addressed

**Comment by parlarjb** [comment]
LGTM, merge on pass

**Pull request merged**